lib/lib-openapi/src/test/java/org/sagebionetworks/translator/ControllerToControllerModelTranslatorTest.java

```diff
@@ -0,0 +1,682 @@
+				.withRequestBody(getExpectedRequestBody()).withResponse(getExpectedResponseModel());
+		expectedMethods.add(expectedMethod);
+		return expectedMethods;
+	}
+
+	private void addAnnotationElementValues(Map<ExecutableElement, AnnotationValue> elementValues, String keyName,
+			Object value) {
+		ExecutableElement key = Mockito.mock(ExecutableElement.class);
+		Name simpleName = Mockito.mock(Name.class);
+		when(key.getSimpleName()).thenReturn(simpleName);
+		when(simpleName.toString()).thenReturn(keyName);
+
+		AnnotationValue annotationValue = Mockito.mock(AnnotationValue.class);
+		when(annotationValue.getValue()).thenReturn(value);
+
+		elementValues.put(key, annotationValue);
+	}
+
+	@Test
+	public void testTranslate() {
+		ControllerModel expectedControllerModel = new ControllerModel().withDisplayName(CONTROLLER_NAME)
+				.withMethods(getExpectedMethods()).withPath("/");
+
+		TypeElement controller = Mockito.mock(TypeElement.class);
+		when(controller.toString()).thenReturn(CONTROLLER_NAME);
+
+		Mockito.doReturn(getExpectedMethods()).when(translator).getMethods(any(List.class), any(DocTrees.class));
+		// call under test
+		assertEquals(expectedControllerModel, translator.translate(controller, Mockito.mock(DocTrees.class)));
+	}
+
+	@Test
+	public void testGetMethods() {
+		DocTrees docTrees = Mockito.mock(DocTrees.class);
+		DocCommentTree mockDocCommentTree = Mockito.mock(DocCommentTree.class);
+		when(docTrees.getDocCommentTree(any(Element.class))).thenReturn(mockDocCommentTree);
+		when(mockDocCommentTree.getBlockTags()).thenReturn(new ArrayList<>());
+		when(mockDocCommentTree.getFullBody()).thenReturn(new ArrayList<>());
+
+		List<Element> enclosedElements = new ArrayList<>();
+		ExecutableElement method = Mockito.mock(ExecutableElement.class);
+		TypeMirror returnType = Mockito.mock(TypeMirror.class);
+		when(returnType.getKind()).thenReturn(TypeKind.INT);
+		when(method.getReturnType()).thenReturn(returnType);
+		Name methodName = Mockito.mock(Name.class);
+		when(method.getSimpleName()).thenReturn(methodName);
+		when(methodName.toString()).thenReturn(METHOD_NAME);
+		when(method.getKind()).thenReturn(ElementKind.METHOD);
+		enclosedElements.add(method);
+
+		// Mock translator method calls
+		Map<String, String> parameterToDescription = new HashMap<>();
+		parameterToDescription.put(PARAM_1_NAME, PARAM_1_DESCRIPTION);
+		Mockito.doReturn(parameterToDescription).when(translator).getParameterToDescription(any(List.class));
+
+		Map<Class, Object> annotationToModel = new HashMap<>();
+		annotationToModel.put(RequestMapping.class,
+				new RequestMappingModel().withOperation(Operation.get).withPath(METHOD_PATH));
+		annotationToModel.put(ResponseStatus.class, new ResponseStatusModel().withStatus(HttpStatus.OK));
+		Mockito.doReturn(annotationToModel).when(translator).getAnnotationToModel(any(List.class));
+
+		JsonSchema expectedRequestBodySchema = new JsonSchema();
```

Comment:
pulling out the schema translator into its own class will reduce what needs to be tested here.